src/compiler.ts

```diff
@@ -6658,6 +6683,37 @@ export class Compiler extends DiagnosticEmitter {
     return expr;
   }
 
+  compileSuperInstantiate(classInstance: Class, argumentExpressions: Expression[], reportNode: Node): ExpressionRef {
+    // traverse to the top-most visible constructor (except the current one)
+    var currentClassInstance: Class | null = classInstance.base;
+    var constructorInstance: Function | null = null;
+    while (currentClassInstance) {
+      constructorInstance = currentClassInstance.constructorInstance;
+      if (constructorInstance) break; // TODO: check visibility
+      currentClassInstance = currentClassInstance.base;
+    }
+
+    // if a constructor is present, allocate the necessary memory for `this` and call it
+    var expr: ExpressionRef;
+    if (constructorInstance) {
```

Comment:
Is it make sense also check constructor's body and if it empty don't call constructor?

Reply:
Might be possible. Actual constructor code is generated code, though, that can contain other instructions like for initializing fields. In this case, though, that won't happen because a non-zero `this` value is provided. Aside from that I'm not too happy with constructor handling as-is, yet (conditional allocate and zero this for self-alloc gives me headaces but is necessary in order to return a custom instance from a ctor, hmm).

Reply:
of course this case should call constructor anyway:
```ts
class Foo {
   a: i32
   b: i32 = this.a
}
```
and probably this:
```ts
class Foo {
   constructor(public a: i32) {}
}
```

Reply:
Not going to optimize this yet as it would conflict with redoing constructors as mentioned below. But it's on TODO.